libshaderc/include/shaderc.h

```diff
@@ -105,6 +111,11 @@ void shaderc_compile_options_release(shaderc_compile_options_t options);
 void shaderc_compile_options_add_macro_definition(
     shaderc_compile_options_t options, const char* name, const char* value);
 
+// Set flags for the rules on which warnings or errors will be issued. This can
+// either be an individual shaderc_msg_rules flag, or a set of flags or'd together
+void shaderc_compile_options_set_message_rules(
```

Comment:
I think you're right: better to add the int version parameter now.  Make it a uint32_t, since version numbers shouldn't be negative (and despite the Google C++ style guide).  (Entails adding #include <stdint.h>)
Each target env (external) spec determines what the numbers mean.